presto-main/src/main/java/com/facebook/presto/server/QueryProgressStats.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.server;
+
+import com.facebook.presto.execution.QueryStats;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.joda.time.DateTime;
+
+import java.util.OptionalDouble;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class QueryProgressStats
+{
+    private final DateTime executionStartTime;
+    private final long elapsedSeconds;
+    private final long queuedSeconds;
+    private final long cpuSeconds;
+    private final long peakMemoryBytes;
+    private final long inputRows;
+    private final long inputBytes;
+    private final OptionalDouble progressPercentage;
+    private final boolean blocked;
+
+    @JsonCreator
+    public QueryProgressStats(
+            @JsonProperty("executionStartTime") DateTime executionStartTime,
+            @JsonProperty("elapsedSeconds") long elapsedSeconds,
+            @JsonProperty("queuedSeconds") long queuedSeconds,
+            @JsonProperty("cpuSeconds") long cpuSeconds,
+            @JsonProperty("peakMemoryBytes") long peakMemoryBytes,
+            @JsonProperty("inputRows") long inputRows,
+            @JsonProperty("inputBytes") long inputBytes,
+            @JsonProperty("blocked") boolean blocked,
+            @JsonProperty("progressPercentage") OptionalDouble progressPercentage)
+    {
+        this.executionStartTime = requireNonNull(executionStartTime, "executionStartTime is null");
+        this.elapsedSeconds = elapsedSeconds;
+        this.queuedSeconds = queuedSeconds;
```

Comment:
I'm wondering whether scheduled and blocked time is useful. @martint is that too much information?